server/aesdsocket.c

```diff
@@ -0,0 +1,341 @@
+/*
+ * @function	:  To handle socket communication
+ *
+ * @param		:   struct addrinfo *res holds the local address for binding,
+ * 					int socket_fd- socket file descriptor and int accept_fd -file descriptor of client
+ * @return		:  NULL
+ *
+ */
+void socket_connect()
+{
+
+	// setting the initial paramters
+	struct addrinfo hints;		  // for getaddrinfo parameters
+	struct addrinfo *res;		  // to get the address
+	struct sockaddr client_add;	  // to get client address
+	socklen_t client_size;		  // size of sockaddr
+	char buff[BUFFER_SIZE] = {0}; // buffer to receive input string
+	memset(buff, 0, BUFFER_SIZE);
+	int data_count = 0; // for counting the data packet bytes
+
+	// Step-1 Initilaizing the  sockaddr using getaddrinfo
+
+	// clear the hints first
+	memset(&hints, 0, sizeof(hints));
+
+	// set all the hint parameters then
+	hints.ai_flags = AI_PASSIVE;
+	hints.ai_family = AF_INET;
+	hints.ai_socktype = SOCK_STREAM;
+
+	printf("Assigning address for socket\n");
+	// getting address in res using getaddrinfo
+	int temp = getaddrinfo(NULL, server_port, &hints, &res);
+	if (temp != 0) // generating error
+	{
+		printf("Error while allocating address for socket\n");
+		syslog(LOG_ERR, "Error while setting socket address= %s. Exiting.", strerror(errno));
+		exit(EXIT_FAILURE);
+	}
+
+	// Step-2 Opening socket
+	printf("Opening socket\n");
+	socket_fd = socket(PF_INET, SOCK_STREAM, 0); // IP family with type SOCK_STREAM and 0 protocol
+	if (socket_fd == -1)						 // generating error
+	{
+		printf("Error: Socket file descriptor not created\n");
+		syslog(LOG_ERR, "Error while setting socket= %s. Exiting.", strerror(errno));
+		freeaddrinfo(res);
+		exit(EXIT_FAILURE);
+	}
+
+	// Step-3 Binding to address
+	printf("Binding socket descriptor to address\n");
+	int temp2 = bind(socket_fd, res->ai_addr, res->ai_addrlen);
+	if (temp2 == -1) // generating error
+	{
+		printf("Error: Binding with address failed\n");
+		syslog(LOG_ERR, "Error while binding socket= %s. Exiting.", strerror(errno));
+		freeaddrinfo(res);
+		exit(EXIT_FAILURE);
+	}
+
+	// All recived data should be stored in the PATH file_data
+	int file_fd = creat(file_data, 0644); // creating file if it doesn't exist with 644 permissions
+	if (file_fd == -1)					  // generating error
+	{
+		printf("Error while creating file \n");
+		syslog(LOG_ERR, "Error: File could not be created!= %s. Exiting...", strerror(errno));
+		exit(EXIT_FAILURE);
+	}
+	close(file_fd);
+
+	// free res when no longer needed
+	freeaddrinfo(res);
+
+	// Step-4 Listening and Accepting packets in a loop
+	// Loop for listening and accepting data from client
+	while (1)
+	{
+
+		output_buffer = (char *)malloc(BUFFER_SIZE); // mallocing output buffer to send data to file
+		if (output_buffer == NULL)
+		{
+			printf("buffer not created\n");
+			exit(EXIT_FAILURE);
+		}
+		memset(output_buffer, 0, BUFFER_SIZE);
+
+		// Listening for client
+		int temp_listen = listen(socket_fd, MAX_BACKLOG);
```

Comment:
@ayswariya088 you shouldn't need `listen` in your while loop, and at least one student found an issue with this at https://discord.com/channels/1062535844696707115/1079159844252176425/1079209265857835069